src/Type/PostObject/Mutation/PostObjectDelete.php

```diff
@@ -0,0 +1,153 @@
+					}
+
+					/**
+					 * Get the ID from the global ID
+					 */
+					$id_parts = Relay::fromGlobalId( $input['id'] );
+
+					/**
+					 * Stop now if a user isn't allowed to delete a post
+					 */
+					if ( ! current_user_can( $post_type_object->cap->delete_post, absint( $id_parts['id'] ) ) ) {
+						// translators: the $post_type_object->graphql_plural_name placeholder is the name of the object being mutated
+						throw new \Exception( sprintf( __( 'Sorry, you are not allowed to delete %1$s', 'wp-graphql' ), $post_type_object->graphql_plural_name ) );
+					}
+
+					// Check if we should force delete or not
+					$force_delete = ( ! empty( $input['forceDelete'] ) && true === $input['forceDelete'] ) ? true : false;
+
+					/**
+					 * Get the post object before deleting it
+					 */
+					$post_before_delete = get_post( absint( $id_parts['id'] ) );
+
+					/**
+					 * If the post is already in the trash, and the forceDelete input was not passed,
+					 * don't remove from the trash
+					 */
+					if ( true !== $force_delete && 'trash' === $post_before_delete->post_status ) {
+						// Translators: the first placeholder is the post_type of the object being deleted and the second placeholder is the unique ID of that object
+						throw new \Exception( sprintf( __( 'The %1$s with id %2$s is already in the trash. To remove from the trash, use the forceDelete input', 'wp-graphql' ), $post_type_object->graphql_single_name, $input['id'] ) );
+					}
+
+					/**
+					 * If there's no post with the input ID, or the post with that ID is of a different post_type
+					 */
+					if ( empty( $post_before_delete ) || $post_before_delete->post_type !== $post_type_object->name ) {
+						// Translators: the first placeholder is the post_type of the object being deleted and the second placeholder is the unique ID of that object
+						throw new \Exception( sprintf( __( 'No %1$s with id %2$s exists', 'wp-graphql' ), $post_type_object->graphql_single_name, $input['id'] ) );
+					}
+
+					/**
+					 * Delete the post
+					 */
+					$deleted = wp_delete_post( $id_parts['id'], $force_delete );
+
+					if ( empty( $deleted ) ) {
+						// Translators: the first placeholder is the post_type of the object being deleted and the second placeholder is the unique ID of that object
+						throw new \Exception( sprintf( __( 'The %1$s with id %2$s was not deleted', 'wp-graphql' ), $post_type_object->graphql_single_name, $input['id'] ) );
+					}
+
+					/**
+					 * If the post was moved to the trash, spoof the object's status before returning it
+					 */
+					$post_before_delete->post_status = ( false !== $deleted && true !== $force_delete ) ? 'trash' : $post_before_delete->post_status;
+
+					/**
+					 * Return the deletedId and the object before it was deleted
+					 */
+					return [
+						'deletedId' => uniqid(),
```

Comment:
yes, it should indeed. I was messing with an idea for returning a unique ID because Graph.Cool also returns a deletedId. . .but upon inspection, their deletedId is just the ID of the object as well. . .I was thinking it was a unique ID for state management purposes, but we already have the clientMutationId for that. . .so I got this updated.